practical_08/convert_miles_km.py

```diff
@@ -0,0 +1,47 @@
+"""
+Convert miles to kilometres
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+
+__author__ = 'Nicholas Williams'
+
+MILES_TO_KM_SCALE = 1.609
+
+
+class MilesKilometreApp(App):
+    """MilesKilometreApp is a Kivy App for converting miles to kilometres"""
+
+    def build(self):
+        """Build the Kivy app from the kv file"""
+        self.title = "Convert Miles to Kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def increment(self, value):
+        """ Increment by "value" """
+        miles = self.root.ids.input_number.text
+        try:
+            miles = float(miles) + value
+            if miles < 0:
+                # Disable negative distances
+                miles = 0
+            self.root.ids.input_number.text = str(miles)
+        except ValueError:
+            pass
```

Comment:
I interpreted this part of the question as if the input isn't a number, then set it to 0. I kept negative numbers, but you could also just set them as the positive versions.